ui/src/variables/selectors/index.tsx

```diff
@@ -67,6 +67,33 @@ export const getValuesForVariable = (
   return get(state, `variables.values.${contextID}.values.${variableID}`)
 }
 
+export const getTypeForVariable = (
+  state: AppState,
+  variableID: string
+): VariableArguments['type'] => {
+  return get(
+    state,
+    `variables.variables.${variableID}.variable.arguments.type`,
+    ''
+  )
+}
+
+interface ArgumentMap {
+  map: {[key: string]: string}
+  list: string[]
+}
+
+export const getArgumentValuesForVariable = <T extends keyof ArgumentMap>(
+  state: AppState,
+  variableID: string
+): ArgumentMap[T] => {
```

Comment:
This feels a bit too clever... took a while to understand how it could possibly be correct. What about returning `VariableArguments` instead and then using type assertions in the calling code? As far as I can tell, the generic return type is a roundabout way of doing a type assertion.

Reply:
sgtm. I was a bit concerned about the cleverness